### ISSUE TITLE
Possibility to set the local environment separately (dev and prod)

### ISSUE THREAD

**Issue by ivodolenc** [nuxt/nuxt.js#8559]
### Is your feature request related to a problem? Please describe.

Hi, the official [documentation](https://nuxtjs.org/docs/2.x/configuration-glossary/configuration-server/) nicely explains how to configure the `server` option and its very useful. But sometimes it can be difficult to work with the default local `development` and `production` environment.

For example, when you run `yarn dev` command, it starts the default `dev` environment at `http://localhost:3000`.

Also, when you run `yarn start`(after yarn generate) command, it starts the default `prod` environment at `http://localhost:3000`.

As you can see, both environments are accessible by default at `http://localhost:3000`. This can lead to problems with the cache or stucked local dev server, and so on.

### Describe the solution you'd like

It would be nice if we could specify both environment separately (possible feature for Nuxt 3).

Maybe something like this:

```js
// nuxt.config.js

export default {
  server: {
    dev: {
      port: 3030, // default: 3000
      host: '0.0.0.0' // default: localhost,
    },
    prod: {
      port: 8080, // default: 8000
      host: '0.0.0.0' // default: localhost,
    }
  }
}
```

### Describe alternatives you've considered

I've created a simple alternative and it works well, but maybe Nuxt can provide the native options for that.

Basically, it checks the current server (`isDev` ? `dev` : `prod`) and sets the environment accordingly.

Here I will attach links to code example and documentation that explains the feature in more detail.

- [Alternative solution](https://github.com/ivodolenc/nuxt-static-boilerplate/blob/master/nuxt/options/server.js)
- [Additional info](https://github.com/ivodolenc/nuxt-static-boilerplate/blob/master/docs/server-configuration.md)

Thanks for Nuxt and support, keep up the amazing work 👍

**Comment by pi0** [comment]
Hi @ivodolenc. Actually you can use `"start": "nuxt start -p 3004 -H 0.0.0.0"`  `package.json` scripts or using `HOST` and `PORT` environment variables.

One alternative would be using [config](https://www.npmjs.com/package/config) package that allows setting config per environment and exposes a merged interface.

Closing this since there is more than one way loading/setting envionment specific configuration.

**Comment by ivodolenc** [comment]
@pi0  Thanks for replying
